File: src/main/java/by/architecture_map/belarus/entity/ConstructionImage.java
package by.architecture_map.belarus.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrimaryKeyJoinColumn;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@PrimaryKeyJoinColumn(name = "image_id")
@Data
@EqualsAndHashCode(callSuper = true)
public class ConstructionImage extends Image {

    @JsonIgnoreProperties("images")
    @ManyToOne
    @JoinColumn(name = "construction_id")
    private Construction construction;

    private String takenTime;
}

File: src/main/kotlin/by/architecture_map/belarus/entity/ConstructionImage.kt
package by.architecture_map.belarus.entity

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.PrimaryKeyJoinColumn

@Entity
@PrimaryKeyJoinColumn(name = "image_id")
data class ConstructionImage(

    @JsonIgnoreProperties("images")
    @ManyToOne
    @JoinColumn(name = "construction_id")
    var construction: Construction? = null,
    var takenTime: String? = null

) : Image()
